test/index.js

```diff
@@ -6,68 +6,98 @@ import {resolve}    from 'path';
+      file: file,
+      importer: jsonImporter,
+      ...opts
     });
+  }
+}
 
+function testExpectedCSS(result) {
     expect(result.css.toString()).to.eql(EXPECTATION);
+}
+
+
+describe('Import type test', function() {
+
+  it('imports strings', function() {
+    ['./test/fixtures/strings/style.scss',
+      './test/fixtures/strings/style-js.scss'
+    ].map(sassRenderFile()).forEach(testExpectedCSS);
   });
 
   it('imports lists', function() {
-    let result = sass.renderSync({
-      file: './test/fixtures/lists/style.scss',
-      importer: jsonImporter
-    });
-
-    expect(result.css.toString()).to.eql(EXPECTATION);
+    ['./test/fixtures/lists/style.scss',
+      './test/fixtures/lists/style-js.scss'
+    ].map(sassRenderFile()).forEach(testExpectedCSS);
   });
 
   it('imports maps', function() {
-    let result = sass.renderSync({
-      file: './test/fixtures/maps/style.scss',
-      importer: jsonImporter
-    });
-
-    expect(result.css.toString()).to.eql(EXPECTATION);
+    ['./test/fixtures/maps/style.scss',
+      './test/fixtures/maps/style-js.scss'
+    ].map(sassRenderFile()).forEach(testExpectedCSS);
   });
 
   it('finds imports via includePaths', function() {
-    let result = sass.renderSync({
-      file: './test/fixtures/include-paths/style.scss',
-      includePaths: ['./test/fixtures/include-paths/variables'],
-      importer: jsonImporter
-    });
-
-    expect(result.css.toString()).to.eql(EXPECTATION);
+    ['./test/fixtures/include-paths/style.scss',
+      './test/fixtures/include-paths/style-js.scss'
+    ].map(sassRenderFile({
+      includePaths: ['./test/fixtures/include-paths/variables']
+    })).forEach(testExpectedCSS);
   });
 
   it('finds imports via multiple includePaths', function() {
-    let result = sass.renderSync({
-      file: './test/fixtures/include-paths/style.scss',
-      includePaths: ['./test/fixtures/include-paths/variables', './some/other/path/'],
-      importer: jsonImporter
-    });
+    ['./test/fixtures/include-paths/style.scss',
+      './test/fixtures/include-paths/style-js.scss'
+    ].map(sassRenderFile({
+      includePaths: ['./test/fixtures/include-paths/variables', './some/other/path/']
+    })).forEach(testExpectedCSS);
+  });
 
-    expect(result.css.toString()).to.eql(EXPECTATION);
+  it(`strips non-valid JSON values from JS exports`, function() {
+    function render() {
+      sass.renderSync({
+        file: './test/fixtures/non-valid-json/style-js.scss',
+        importer: jsonImporter
+      });
+    }
+    expect(render).to.throw(`Undefined variable: "$color-red".`)
   });
 
-  it(`throws when an import doesn't exist`, function() {
+  it(`throws when importing non-valid JSON from JS exports`, function() {
     function render() {
       sass.renderSync({
-        file: './test/fixtures/include-paths/style.scss',
-        includePaths: ['./test/fixtures/include-paths/foo'],
+        file: './test/fixtures/wrong-js-export/style-js.scss',
```

Comment:
What's the difference between this test and the one above?

The "non-valid" test has fixtures in the "wrong-js-export" folder, while the test above it has fixtures in "non-valid". It's a bit confusing.